llvm-runtimes/libgcc/libgcc-20.0.0.9999.ebuild

```
@@ -0,0 +1,153 @@
+	)
+"
+
+LLVM_COMPONENTS=( compiler-rt cmake llvm/cmake llvm-libgcc )
+LLVM_TEST_COMPONENTS=( llvm/include/llvm/TargetParser )
+llvm.org_set_globals
+
+python_check_deps() {
+	use test || return 0
+	python_has_version ">=dev-python/lit-15[${PYTHON_USEDEP}]"
+}
+
+pkg_setup() {
+	if target_is_not_host || tc-is-cross-compiler ; then
+		# strips vars like CFLAGS="-march=x86_64-v3" for non-x86 architectures
+		CHOST=${CTARGET} strip-unsupported-flags
+		# overrides host docs otherwise
+		DOCS=()
+	fi
+	python-any-r1_pkg_setup
+}
+
+src_configure() {
+	# We need to build a separate copy of compiler-rt, because we need to disable the
+	# COMPILER_RT_BUILTINS_HIDE_SYMBOLS option - compatibility with libgcc requires
+	# visibility of all symbols.
+
+	llvm_prepend_path "${LLVM_MAJOR}"
+
+	# LLVM_ENABLE_ASSERTIONS=NO does not guarantee this for us, #614844
+	use debug || local -x CPPFLAGS="${CPPFLAGS} -DNDEBUG"
+
+	local -x CC=${CTARGET}-clang CXX=${CTARGET}-clang++
+	strip-unsupported-flags
+
+	local mycmakeargs=(
+		-DCOMPILER_RT_INSTALL_PATH="${EPREFIX}/usr/lib/clang/${LLVM_MAJOR}"
+
+		-DCOMPILER_RT_INCLUDE_TESTS=$(usex test)
+		-DCOMPILER_RT_BUILD_CTX_PROFILE=OFF
+		-DCOMPILER_RT_BUILD_LIBFUZZER=OFF
+		-DCOMPILER_RT_BUILD_MEMPROF=OFF
+		-DCOMPILER_RT_BUILD_ORC=OFF
+		-DCOMPILER_RT_BUILD_PROFILE=OFF
+		-DCOMPILER_RT_BUILD_SANITIZERS=OFF
+		-DCOMPILER_RT_BUILD_XRAY=OFF
+
+		-DCOMPILER_RT_BUILTINS_HIDE_SYMBOLS=OFF
+
+		-DPython3_EXECUTABLE="${PYTHON}"
+	)
+
+	if use amd64 && ! target_is_not_host; then
+		mycmakeargs+=(
+			-DCAN_TARGET_i386=$(usex abi_x86_32)
+			-DCAN_TARGET_x86_64=$(usex abi_x86_64)
+		)
+	fi
+
+	if use test; then
+		mycmakeargs+=(
+			-DLLVM_EXTERNAL_LIT="${EPREFIX}/usr/bin/lit"
+			-DLLVM_LIT_ARGS="$(get_lit_flags)"
+
+			-DCOMPILER_RT_TEST_COMPILER="${EPREFIX}/usr/lib/llvm/${LLVM_MAJOR}/bin/clang"
+			-DCOMPILER_RT_TEST_CXX_COMPILER="${EPREFIX}/usr/lib/llvm/${LLVM_MAJOR}/bin/clang++"
+		)
+	fi
+
+	cmake_src_configure
+}
+
+# Returns the path to the locally built compiler-rt shared library.
+get_rtlib() {
+	local rtlib_filename=$(basename $($(tc-getCC) -print-libgcc-file-name))
+	echo "${BUILD_DIR}/lib/linux/${rtlib_filename}"
+}
+
+src_compile() {
+	local -x CC=${CTARGET}-clang CXX=${CTARGET}-clang++
+	local libdir=$(get_libdir)
+
+	cmake_src_compile
+
+	# Use the llvm-libgcc's version script to produce libgcc.{a,so}, which
+	# combines compiler-rt and libunwind into a libgcc replacement.
+	#
+	# What we do here is similar to what upstream does[0], with the following
+	# differences:
+	#
+	# * We build the local copy of compiler-rt manually, to have a full control
+	#   over CMake options.
+	# * Upstream links the locally built copy of libunwind statically. We link the
+	#   system-wide libunwind dynamically.
+	#
+	# [0] https://github.com/llvm/llvm-project/blob/llvmorg-19.1.7/llvm-libgcc/CMakeLists.txt#L102-L120
+	$(tc-getCPP) \
+		"${WORKDIR}/llvm-libgcc/gcc_s.ver.in" \
+		-o gcc_s.ver || die
+	$(tc-getCC) -nostdlib \
```

```suggestion
	${CC} -nostdlib \
```

`tc-getCC` makes sense when you needed a fallback but since we're forcing `CC`…

---

Done